checkers-backend/src/api/game/game.controller.ts

```diff
@@ -0,0 +1,73 @@
+import { Controller, Get, Post, Param, Body, Put, Delete } from '@nestjs/common';
+import { ApiService } from '../api.service';
+import { Datastore } from '@google-cloud/datastore';
+import { take } from 'rxjs';
+
+@Controller('api/v1/Game')
+export class GameController {
+    constructor(private readonly apiService: ApiService) {}
+    datastore = new Datastore({databaseId:'checkers-datastore', projectId:"checkers-zsl"});
+
+    @Get()
+    async FindAll(): Promise<object> {
+        try
+        {
+            const query = this.datastore.createQuery('Game')
+            const result = await this.datastore.runQuery(query)
+            return await this.apiService.ApiSuccessData({result})
+        }
+        catch(err){return await this.apiService.ApiErrorData(err);}
+    }
+    
+    // @Get(":id")
+    // async FindOne(@Param() id: number ): Promise<object> {
+    //     try{
+    //         const taskKey = await this.datastore.key([ 'Game', id ]);
+    //         const result = await this.datastore.createQuery('Game').hasAncestor(taskKey);
+    //         return await this.apiService.ApiSuccessData({result})
+    //     }
+    //     catch(err){return await this.apiService.ApiErrorData(err);}
+    // }
+
+    @Post()
+    async AddOne(@Body() body: object): Promise<string> {
```

Comment:
it's good idea to create dto classes -> https://medium.com/@yelinliu/dto-explained-in-nestjs-3a296498d77b

Reply:
or just interfaces or even types
https://www.typescriptlang.org/docs/handbook/interfaces.html
https://www.typescriptlang.org/docs/handbook/advanced-types.html